woocommerce/woocommerce-android
# Domain change: Analytics

Fixes #8433, a subtask #8238. 

The PR adds multiple analytics declared in pe5sF9-15a#comment-1710.

**To test:**

Domain registration using webview purchase
1. Login using an admin account (give yourself free credits first in SA)
2. Select a store that's _not_ eligible for a free domain 
3. Go to Settings -> Domains
4. Verify that `🔵 Tracked: settings_domains_tapped` is logged
5. Verify that `🔵 Tracked: custom_domains_step, Properties: {"source":"settings","step":"dashboard"...` is logged
6. Tap on Add domain button
7. Verify that `🔵 Tracked: custom_domains_step, Properties: {"source":"settings","step":"picker"...` is logged
8. Select a domain & tap Continue
9. Verify that `🔵 Tracked: custom_domains_step, Properties: {"source":"settings","step":"web_checkout"...` is logged
10. Finish the purchase and wait for completion
12. When the success screen is shown, verify that `🔵 Tracked: custom_domains_step, Properties: {"source":"settings","step":"purchase_success"...` is logged
13. Verify that `🔵 Tracked: custom_domain_purchase_success, Properties: {"source":"settings","use_domain_credit":false...` is logged

Domain registration using free credit
1. Login using an admin account
2. Select a store that's eligible for a free domain (or create a new one from the app)
3. Go to Settings -> Domains
4. Tap on Claim free domain banner button
5. Find a non-premium domain that's free for the first year
6. Tap on Continue
7. Verify that `🔵 Tracked: custom_domains_step, Properties: {"source":"settings","step":"contact_info"...` is logged
8. Try changing the e-mail address to some invalid value and tap Register
9. Verify that `🔵 Tracked: domain_contact_info_validation_failed, Properties: {"source":"settings","error_context":"FreeDomainRegistrationViewModel","error_type":"EMAIL","error_description":"There was an error validating your contact information. The field \"Email\" is not valid."` is logged
10. Fix the email address
11. Turn on the Flight mode
12. Tap on Register
13. Verify that `🔵 Tracked: custom_domain_purchase_failed, Properties: {"source":"settings","use_domain_credit":true...` is logged
14. Turn off the Flight mode
15. Tap on Register
16. When the success screen is shown, verify that `🔵 Tracked: custom_domains_step, Properties: {"source":"settings","step":"purchase_success"...` is logged
17. Verify that `🔵 Tracked: custom_domain_purchase_success, Properties: {"source":"settings","use_domain_credit":true...` is logged
